src/machine/mod.rs

```diff
@@ -0,0 +1,34 @@
+pub mod behavior;
+pub mod state;
+
+use self::state::State;
+
+pub struct Machine {
+    _state: State,
+    /* TODO add more fields... */
+}
+
+impl Machine {
+    // TODO FYI: Machine uses the builder pattern.
+
+    pub fn new() -> Self {
+        unimplemented!();
+    }
+
+    pub fn load(self, _pos: u64, _data: &[u8]) -> Self {
+        // Loads the slice data to the position pos in memory
+        unimplemented!();
+    }
+
+    pub fn set_breakpoint(self, _bp: u64) -> Self {
```

Comment:
Breakpoints are places in the program where the execution will pause, so that the programmer can examine the state of the machine at that point. Yes, it's usually part of debugging.